src/backend/distributed/utils/colocation_utils.c

```diff
@@ -240,6 +241,12 @@ CreateColocationGroupForRelation(Oid sourceRelationId)
 static void
 MarkTablesColocated(Oid sourceRelationId, Oid targetRelationId)
 {
+	if (IsCitusLocalTable(sourceRelationId) || IsCitusLocalTable(targetRelationId))
```

Comment:
Very Minor: Maybe create a method `EnsureNotCitusLocalTable(relationId)` and use as `EnsureNotCitusLocalTable(sourceRelationId); EnsureNotCitusLocalTable(targetRelationId);`

Reply:
But in this context, don't we want to give the error message specific for the colocation ?